### PR TITLE
Support for <equation>s

### PR DESCRIPTION
**Pull request by aiwenar**

### CODE REVIEW

File: src/normalize/equation.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    const [node, path] = entry
+
+    if (Equation.isEquation(node)) {
+        // Equations must not be empty
+        if (node.children.length === 0) {
+            Transforms.removeNodes(editor, { at: path })
+            return true
+        }
+
+        // Rules must only have a single child
```
Comment (jarosik10) [inline src/normalize/equation.ts:20]: Rules?

File: src/plugin.ts [hunk 2]
```diff
@@ -8,8 +8,9 @@ import normalizeNode from './normalize'
 import { isInline, isVoid } from './interfaces'
 
 /** A Slate editor augmented with support for editing CNX documents. */
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface CnxEditor extends Editor {
+    /** Can {@code element} be used as content of an {@link Equation}? */
+    isEquationContent(element: Element): boolean
```
Comment (jarosik10) [inline src/plugin.ts:13]: This comment doesn't seems really clear to me.
btw. `<equation>` can be directly followed be `<math>`, right?
```html
<equation>
   <math>...</math>
</equation>
```

File: src/normalize/equation.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        if (node.children.length > 1) {
+            Transforms.splitNodes(editor, { at: [...path, node.children.length - 1] })
+            return true
+        }
+
+        // Equation can only contain elements from a predefined set and in
+        // a strict order.
+        if (!Element.isElement(node.children[0]) || !editor.isEquationContent(node.children[0])) {
```
Comment (jarosik10) [inline src/normalize/equation.ts:28]: should it be only checked for the equation's first child?

**Review by jarosik10 — CHANGES_REQUESTED** [review]

File: src/normalize/equation.ts (hunk 3 above)
Comment (aiwenar) [reply src/normalize/equation.ts:28]: The previous `if` enforces that equations only have one child.

**Review by jarosik10 — APPROVED** [review]
Looks good:) please squash the fixups

**Pull request merged**